frontend/src/Store/Actions/Creators/createHandleActions.js

```diff
@@ -42,6 +42,7 @@ export default function createHandleActions(handlers, defaultState, section) {
 
         if (_.isArray(payload.data)) {
           newState.items = payload.data;
+          newState.itemMap = _.zipObject(_.map(payload.data, 'id'), _.range(payload.data.length));
```

Comment:
I think this could be replaced by:

```
newState.itemMap = payload.data.reduce((acc, d, index) => {
  acc[d.id] = index;
  return acc;
}, {});
```

As a helper function in this file (since I don't see a huge need outside of here) it'd eliminate the need for 3 different lodash functions and whatever iterating they need to do.